app/viewModels/houseConsignment/index/documents/AddAnotherHouseConsignmentDocumentViewModel.scala

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package viewModels.houseConsignment.index.documents
+
+import config.FrontendAppConfig
+import controllers.houseConsignment.index.documents.routes
+import models.DocType.Previous
+import models.removable.Document
+import models.{ArrivalId, HouseConsignmentLevelDocuments, Index, Mode, RichOptionalJsArray, UserAnswers}
+import pages.houseConsignment.index.documents.TypePage
+import pages.sections.houseConsignment.index.documents.DocumentsSection
+import play.api.i18n.Messages
+import play.api.libs.json.JsArray
+import play.api.mvc.Call
+import viewModels.documents.Documents
+import viewModels.{AddAnotherViewModel, ListItem}
+
+case class AddAnotherHouseConsignmentDocumentViewModel(
+  listItems: Seq[ListItem],
+  onSubmitCall: Call,
+  nextIndex: Index,
+  documents: HouseConsignmentLevelDocuments,
+  allowMore: Boolean
+) extends AddAnotherViewModel {
+
+  override val prefix: String = "houseConsignment.index.document.addAnotherDocument"
+
+  override def maxCount(implicit config: FrontendAppConfig): Int = config.maxSupportingDocumentsHouseConsignment + config.maxTransportDocumentsHouseConsignment
+
+  def maxLimitLabel(houseConsignmentIndex: Index)(implicit messages: Messages): String =
+    messages(s"$prefix.maxLimit.label", houseConsignmentIndex)
+
+  def title(houseConsignmentIndex: Index)(implicit messages: Messages): String =
+    messages(s"$prefix.$emptyOrSingularOrPlural.title", count, houseConsignmentIndex)
+
+  def heading(houseConsignmentIndex: Index)(implicit messages: Messages): String =
+    messages(s"$prefix.$emptyOrSingularOrPlural.heading", count, houseConsignmentIndex)
+
+  def legend(houseConsignmentIndex: Index)(implicit messages: Messages): String = count match {
```

Comment:
I believe you can use the already existing legend method in the AddAnotherViewModel.

Reply:
This one is slightly different, it passes in the houseIndex. Similar the [items one.](https://github.com/hmrc/manage-transit-movements-unloading-frontend/blob/main/app/viewModels/houseConsignment/index/items/document/AddAnotherHouseConsignmentDocumentViewModel.scala#L53)